common/web/keyboard-processor/src/text/keyboardProcessor.ts

```diff
@@ -41,8 +42,9 @@ interface EventMap {
 
 export default class KeyboardProcessor extends EventEmitter<EventMap> {
   public static readonly DEFAULT_OPTIONS: ProcessorInitOptions = {
-    baseLayout: 'us'
-  }
+    baseLayout: 'us',
+    defaultOutputRules: new DefaultOutput()
```

Comment:
Right, that.  I kind of expected that question.  This goes back to how we've been handling the differentiated behavior for the 'tab' and 'enter' keys - but mostly the 'tab' key.

For `<input>` elements in the browser, 'tab' and 'enter' do not output text by default - they perform browser-default (or default-emulating) navigation.  This also goes for 'tab' in `<textarea>` elements.

For WebView-embedded KMW, like in the Android app, 'tab' should generally output _the character_ (`\t`) instead of performing navigation.

In namespaced KMW, we relied upon prototype extension (i.e. monkey-patching?) to manage aspects of the behavioral difference - see https://github.com/keymanapp/keyman/blob/master/web/src/engine/main/dom/domDefaultOutput.ts for the file that does this.

Relevant snippets:

https://github.com/keymanapp/keyman/blob/f0cc2b56e64acd0c35256e7f49587ea4c18b094c/web/src/engine/main/dom/domDefaultOutput.ts#L7-L9

https://github.com/keymanapp/keyman/blob/f0cc2b56e64acd0c35256e7f49587ea4c18b094c/web/src/engine/main/dom/domDefaultOutput.ts#L24-L44

For namespaced code, this... well, it at least _worked_.  Keep in mind that `domManager` does not exist headlessly and thus cannot be accessed by `keyboard-processor`.  As a result, the second block above could not have been included within it; I made the split there previously and relied on the ol' "triage it until later" strategy.  Well... it's "later."

That has always been one of my "least-happy" solutions to things in KMW 12.0 (when `keyboard-processor` was spun off).  It did the job, but it still felt a bit dirty.  But, at least the namespacing structure helped make clear exactly what was happening... or at least, it did in my mind.

----

I _could_ try to do a similar strategy now, but the pattern feels far murkier with modules - if my understanding is correct, we'd be monkey-patching our own modules.  In all honesty, it _probably_ would work fine, but that felt far too murky to sustain with the new organizational pattern.  I've come back to it multiple times now, and I'm only now becoming relatively comfortable with the idea of maintaining it in that pattern.

What the strategy essentially served to do in the past... was to create an "informal subclass" specific to the main Keyman Engine for Web build products that that of the lower-level keyboard-processor.  It inherits all base `DefaultOutput` behavior and then overrides a method or two with extended functionality.  

Rather than maintain the "informal inheritance" pattern of monkey patching... I just replaced it with formal inheritance.  So... this simply _formalizes_ the old pattern and makes it more explicit.  This accomplishes our goals here without relying on the monkey-patch pattern, which is something I believe is generally avoided when possible these days.

Reply:
OK, so the strategy sounds solid. But I want to pick up on one thing you wrote:

> For WebView-embedded KMW, like in the Android app, 'tab' should generally output the character (\t) instead of performing navigation.

I don't think this is correct. For Android app, tab should generally be passed to the app, where it can decide to emit a tab character or move to another element. It's not a critical issue, because generally, this is only relevant for external hardware keyboards, and many apps won't do anything with it anyway, but I don't think we should be emitting `\t` by default.